app/rdf/query-cube-termsets.ts

```diff
@@ -0,0 +1,39 @@
+import ParsingClient from "sparql-http-client/ParsingClient";
+
+import { Termset } from "@/domain/data";
+import { buildLocalizedSubQuery } from "@/rdf/query-utils";
+
+export const getCubeTermsets = async (
+  iri: string,
+  options: {
+    locale: string;
+    sparqlClient: ParsingClient;
+  }
+): Promise<Termset[]> => {
+  const { sparqlClient, locale } = options;
+  const qs = await sparqlClient.query.select(
+    `PREFIX cube: <https://cube.link/>
+PREFIX meta: <https://cube.link/meta/>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX schema: <http://schema.org/>
+PREFIX sh: <http://www.w3.org/ns/shacl#>
+
+SELECT DISTINCT ?termsetIri ?termsetLabel WHERE {
+    VALUES (?iri) {(<${iri}>)}
+
+    ?iri cube:observationConstraint/sh:property ?dimension .
+    ?dimension a cube:KeyDimension .
+    ?dimension sh:in/rdf:rest*/rdf:first ?value.
+    ?value schema:inDefinedTermSet ?termsetIri .
+    ?termsetIri a meta:SharedDimension .
```

Comment:
In the end, the query will evolve in the end so that we do not have to check each dimension values to find termsets. The dimension to termsets triplets will be directly materialized, so that querying can be faster.